src/main/java/de/tum/in/www1/artemis/service/CourseScoreCalculationService.java

```diff
@@ -31,6 +33,11 @@ public class CourseScoreCalculationService {
 
     private static final double SCORE_NORMALIZATION_VALUE = 0.01;
 
+    /**
+     * Scores are calculated for all exercises ("total") as well as for the subset of exercises of each exercise type.
```

Comment:
The same `calculateCourseScoreForStudent` function is called multiple times, once with all exercises to calculate the `TOTAL` scores and then once for each `ExerciseType` with only the exercises that belong to that `ExerciseType`. So my first idea was to just add `TOTAL` into that `ExerciseType` enum but that would be a little ugly because there are no exercises in the end that you can actually retrieve by that `TOTAL` `ExerciseType`. The String here is only used to add the total scores into the `scoresPerExerciseType` Map that is sent to the client: `scoresPerExerciseType.put(TOTAL, totalScores)`, so right now I could also just replace the `TOTAL` there with `"total"`. Any preferences?